File: tools/ghidra/ExportSymbols.java
import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.Function;
import ghidra.program.model.listing.FunctionManager;
import java.io.PrintWriter;
import java.io.FileWriter;

public class ExportSymbols extends GhidraScript {

    @Override
    public void run() throws Exception {
        String[] args = getScriptArgs();
        String out = args.length > 0 ? args[0] : "symbols.csv";

        FunctionManager fm = currentProgram.getFunctionManager();
        StringBuilder sb = new StringBuilder("address,size,name\n");
        int count = 0;
        long total = 0;

        for (Function f : fm.getFunctions(true)) {
            long size = f.getBody().getNumAddresses();
            sb.append("0x").append(f.getEntryPoint()).append(',')
              .append(size).append(',')
              .append(f.getName()).append('\n');
            count++;
            total += size;
        }

        try (PrintWriter pw = new PrintWriter(new FileWriter(out))) {
            pw.print(sb);
        }
        println("exportadas " + count + " funcoes (" + total +
                " bytes) -> " + out);
    }
}

File: tools/ghidra/export_symbols.py
# @category Dash
# Exporta funcoes analisadas para symbols.csv (address,size,name)
# Args: <caminho do csv de saida>
import os

args = getScriptArgs()
out_path = args[0] if args else "symbols.csv"

fm = currentProgram.getFunctionManager()
listing = currentProgram.getListing()

lines = ["address,size,name"]
count = 0
total_bytes = 0
for f in fm.getFunctions(True):
    body = f.getBody().getNumAddresses()
    lines.append("%s,%d,%s" % (f.getEntryPoint(), body, f.getName()))
    count += 1
    total_bytes += body

with open(out_path, "w") as fh:
    fh.write("\n".join(lines) + "\n")

print("exportadas %d funcoes (%d bytes) -> %s" % (count, total_bytes, out_path))
